python/HH_SF.py

```diff
@@ -61,4 +61,78 @@ def HHbtag_SFRDF(**kwargs):
 
     """
     year = kwargs.pop("year")
-    return lambda: HHbtag_SFRDFProducer(year, **kwargs)
\ No newline at end of file
+    return lambda: HHbtag_SFRDFProducer(year, **kwargs)
+
+class HH_pNetSFRDFProduced(JetLepMetSyst):
+    def __init__(self, *args, **kwargs):
+        super(HH_pNetSFRDFProduced, self).__init__(*args, **kwargs)
+        self.isMC = kwargs.pop("isMC")
+        self.year = kwargs.pop("year")
+
+        # Check whether HH-like, TT-like, DY-like or other for pNet SF:
+        #  - HH-like : ggHH, VBFHH, ZH, WH, ttH, ggH, qqH, ttWH, ttZH
+        #  - TT-like : TT fullyLep, semiLep, fullyHad
+        #  - DY_like : all DY samples
+        #  - other   : all other samples
+        dsetName = kwargs.pop("dataset")
+        if "HH" in dsetName or "Hto2B" in dsetName or "WH" in dsetName or "ZH" in dsetName:
+            sampleType = "HHlike"
+        elif "TTto" in dsetName:
+            sampleType = "TTlike"
+        elif "DY" in dsetName:
+            sampleType = "DYlike"
+        else:
+            sampleType = "other"
+        self.sampleType = sampleType
+
+        if self.isMC:
+            base = "{}/{}/src/Tools/Tools".format(os.getenv("CMT_CMSSW_BASE"), os.getenv("CMT_CMSSW_VERSION"))
+            ROOT.gROOT.ProcessLine(".L {}/interface/PNetSFInterface.h".format(base))
+
+            ROOT.gInterpreter.Declare("""
+                auto PNetAK8SF = PNetSFInterface("%s");
+            """ % (self.year))
+
+    def run(self, df):
+        # In case of data: no SFs
+        if not self.isMC:
+            return df, []
+
+        # In case of MC: compute SFs based on jet pT, sample type and gen-matches variables
+        df = df.Define("fatjet_pNet_SF_vec", "PNetAK8SF.getSFvec(FatJet_pt{1}, fatjet_JetIdx, "
```

Comment:
As far as I understood in ZH/ZZ you don't (yet) have the different sys flavors of `FatJet_pt`, but maybe it's one of the improvements @tcuisset is working ok?